core/frontend/src/views/CloudView.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+  <v-card
+    elevation="0"
+    class="mx-auto my-12 px-1 py-6 text-center"
+    min-width="370"
+    max-width="720"
+  >
+    <v-card>
+      <v-card-title>Cloud Registration</v-card-title>
+      <strong>Machine ID:</strong>
+      <div> {{ machineid }}</div>
+      <v-card-text />
+    </v-card>
+
+    <v-card>
+      <v-card-title>Cloud Token</v-card-title>
+      <v-card-text>
+        <v-text-field v-model="cloud_token" outlined />
+        <a href="https://app.blueos.cloud/vehicle/register/" taget="_blank">
+          Click here to create a new vehicle in BlueOS Cloud
+        </a>
+      </v-card-text>
+      <br>
+      <v-btn @click="join">
+        Save
+      </v-btn>
+      <v-card-text />
+    </v-card>
+  </v-card>
+</template>
+
+<script>
+import Vue from 'vue'
+
+import bag from '@/store/bag'
+import commander from '@/store/commander'
+
+export default Vue.extend({
+  name: 'CloudView',
+  data() {
+    return {
+      machineid: undefined,
+      cloud_token: '',
+    }
+  },
+  async mounted() {
+    const result = await commander.command("python -c 'import machineid; print(machineid.hashed_id())'")
```

Comment:
Isn't it better to have helper to provide such information on the frontend over doing python over ts ?

Reply:
it definitely is. actually I was thinking it could be done by linux2rest, what do you think?

Reply:
> This ID uniquely identifies the host. It should be considered
       "confidential", and must not be exposed in untrusted
       environments, in particular on the network. If a stable unique
       identifier that is tied to the machine is needed for some
       application, the machine ID or any part of it must not be used
       directly. Instead the machine ID should be hashed with a
       cryptographic, keyed hash function, using a fixed,
       application-specific key. That way the ID will be properly
       unique, and derived in a constant way from the machine ID but
       there will be no way to retrieve the original machine ID from the
       application-specific one. The
       [sd_id128_get_machine_app_specific(3)](https://man7.org/linux/man-pages/man3/sd_id128_get_machine_app_specific.3.html) API provides an
       implementation of such an algorithm.


https://man7.org/linux/man-pages/man5/machine-id.5.html

So I guess we should pass an application key to it, like `machineid.hashed_id({blueos})`

To avoid using commander, we could bind the /etc/machine-id as read-only to the BlueOs container.

Reply:
except we dont want to use machine-id, right? since it does a prettye bad job at "uniquely identifying the host"...
I vote that at first we just hash the cpu serial. I have no strong feelings about that being confidential.

For it do actually be confidention, we'd have to implement the "join" logic mostly in the backend, though

Reply:
I agree with pascal about the security concerns.
For now, what I can point is:

For a rasp we can use the serial number, and for everything else we could have custom support or use directly https://docs.rs/machineid-rs/1.2.4/machineid_r, that should support everything necessary for hardware identification generation and add it in linux2rest as pointed by @Williangalvani for easy access API. For more information about crates capabilities, check https://docs.rs/machineid-rs/1.2.4/machineid_rs/enum.HWIDComponent.html

But, that may not fix the issue, since we are going to have only a local source that may not work reliable.

I vote for us to have an automatic request by blueos to blueos.cloud requesting an unique identifier based on the HARDWARE_IDENTIFIER.

```
HARDWARE_IDENTIFIER = `${HARDWARE_IDENTIFIER}-${LINUX_MACHINE_ID}`
BLUEOS_CLOUD_IDENTIFIER = request_blueos_cloud_identifier(HARDWARE_IDENTIFIER)
BLUEOS_IDENTIFIER = `${BLUEOS_CLOUD_IDENTIFIER}-${HARDWARE_IDENTIFIER}`
```

With that we ensure a global unique identifier.

Reply:
`HARDWARE_IDENTIFIER` is generated during execution time, so with that we're going to check if the two computers share the same CLOUD unique identifier with different hardware identification. That'll happen on the case where the sdcard is cloned. 

How `HARDWARE_IDENTIFIER` will be created depend on us. Right now, I would use the pi serial number + `machineid-rs (SystemID, CPUCores, MacAddress, CPUID)`, where the `machineid-rs` would be the standard way of doing it if there are no custom support for the identified system.

Reply:
> `HARDWARE_IDENTIFIER` is generated during execution time, so with that we're going to check if the two computers share the same CLOUD unique identifier with different hardware identification. That'll happen on the case where the sdcard is cloned.
> 
> How `HARDWARE_IDENTIFIER` will be created depend on us. Right now, I would use the pi serial number + `machineid-rs (SystemID, CPUCores, MacAddress, CPUID)`, where the `machineid-rs` would be the standard way of doing it if there are no custom support for the identified system.

Okay, so you want to do only (B)?

Reply:
> > `HARDWARE_IDENTIFIER` is generated during execution time, so with that we're going to check if the two computers share the same CLOUD unique identifier with different hardware identification. That'll happen on the case where the sdcard is cloned.
> > How `HARDWARE_IDENTIFIER` will be created depend on us. Right now, I would use the pi serial number + `machineid-rs (SystemID, CPUCores, MacAddress, CPUID)`, where the `machineid-rs` would be the standard way of doing it if there are no custom support for the identified system.
> 
> Okay, so you want to do only (B)?

It's `C`, where we keep track of both, we ask `BLUEOS_CLOUD_IDENTIFIER` that is always unique. With this, we are going to be sure that is always a single unique identifier per vehicle.

# SD Card cloned

The `HARDWARE_IDENTIFIER` will not be the same when comparing to `BLUEOS_CLOUD_IDENTIFIER` on the cloud database, we are going to detect this case.

# New SD Card, old registered hardware

The `HARDWARE_IDENTIFIER` will match but `BLUEOS_CLOUD_IDENTIFIER` not once connects to `blueos.cloud`. We are going to detect this case as well.